### ISSUE TITLE
Wallets can emit `'objectiveStarted'` multiple times for the same objective

### ISSUE THREAD

**Issue by andrewgordstewart** [statechannels/statechannels#3289]
**Describe the bug**
From [here](https://github.com/statechannels/statechannels/pull/3281/files#diff-93ef0bace6dd5a6424f1f5ba77514bc2788c66488a71ad89b424b3761491dbe9R844-R846)

```
    // HACK (1): This may cause the wallet to re-emit `'objectiveStarted'` multiple times
    // For instance, a peer who sends me an objective `o`, and then triggers `syncObjectives`
    // including `o`, will cause my wallet to emit `'objectiveStarted'` for `o` twice.
```

**To Reproduce**
Run
```
yarn test src/__test__/sync-objectives.test.ts -t 'even if'
```

on 7e2cd1de390aa76e4eca8a278a1bf1c20d1a5d70, which is in `master`.

**Expected behavior**
If Bob stores objective `o`, and Alice sends Bob a message containing `o`, then Bob's wallet should not emit `'objectiveStarted'`.

**Additional context**
While writing https://github.com/statechannels/statechannels/pull/3281, I expected that `ensureObjective` would only return an objective if the objective was not yet stored. This is not true -- it returns the objective no matter if it's stored or not.

I intentionally left the behaviour as is in #3281, to avoid the PR from becoming complex. I did not foresee the conflict between that behaviour and the test added in https://github.com/statechannels/statechannels/pull/3275 which is now failing.

One resolution is for `ensureObjective` to indicate whether or not the wallet was already storing the objective. Then, the wallet can conditionally place it into `createdObjectives`.

**Comment by lalexgap** [comment]
This is blocked by https://github.com/statechannels/statechannels/pull/3313#pullrequestreview-592770505 .

**Comment by andrewgordstewart** [comment]
Closed by #3541
